hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/CloudObjectsSelectorCommon.java

```diff
@@ -218,6 +221,14 @@ public static String generateFilter(Type type,
     if (ignoreRelativePathSubStr.isPresent()) {
       filter.append(SPACE_DELIMTER).append(String.format("and %s not like '%%%s%%'", objectKey, ignoreRelativePathSubStr.get()));
     }
+    if (selectRelativePathRegex.isPresent()) {
+      String prefix = selectRelativePathPrefix.orElse("");
+      String regex = selectRelativePathRegex.get();
+
+      String updatedPathRegex = prefix.isEmpty() || prefix.endsWith(PATH_DELIMITER) ? prefix + regex : prefix + PATH_DELIMITER + regex;
+
+      filter.append(SPACE_DELIMTER).append(String.format("and %s rlike '%s'", objectKey, updatedPathRegex));
```

Comment:
Could this filter be combined with `if (selectRelativePathPrefix.isPresent())` branch above to avoid repeated filtering?

Reply:
Have combined this with prefix clause.